File: parasmittal/Day 17/Largest subarray with 0 sum.cpp
int maxLen(int arr[], int n)
{
    // Your code here
    int sum=0;
    int maxlen=0;
    unordered_map<int,int> um;
    for(int i=0;i<n;i++){
        sum+=arr[i];
        if(sum==0){
            maxlen=i+1;
            continue;
        }
        if(um.find(sum)!=um.end()){
            maxlen=max(maxlen,i-um[sum]);
        }
        else
        um[sum]=i;
        
    }
    return maxlen;
}

File: parnamondal/Week -4/Day -1/Largest subarray with 0 sum .py
# Time Complexity - O(N)
# Space Complexity - O(N)

def maxLen(n, arr):
    s=0
    p={}
    p[0]=-1
    l=0
    m=0
    for i in range (n):
        s+=arr[i]
        if (s in p):
            l=i-p[s]
            m=max(l,m)
        else :
            p[s]=i
    return m
            
